lib/evaluateFunctions.js

```diff
@@ -96,4 +99,421 @@ function absoluteValue(node) {
   }
 }
 
+// Evaluate nthRoot() function.
+// Returns a NodeStatus object.
+function nthRoot(node) {
+  if (!NodeType.isFunction(node, 'nthRoot')) {
+    return NodeStatus.noChange(node);
+  }
+
+  const radicandNode = node.args[0];
+  if (NodeType.isOperator(radicandNode)) {
+    if (radicandNode.op === '^') {
+      return nthRootExponent(node);
+    }
+    else if (radicandNode.op === '*') {
+      return nthRootMultiplication(node);
+    }
+  }
+  else if (NodeType.isConstant(radicandNode)) {
+    return nthRootConstant(node);
+  }
+
+  return NodeStatus.noChange(node);
+}
+
+// Returns the nthRoot evaluated for an exponent node. Expects an exponent under
+// the radicand. Cancels the root and the exponent if possible. Three cases:
+// equal: nthRoot(2^x, x) = 2
+// root > exponent: nthRoot(x^2, 4) = nthRoot(x, 2)
+// exponent > root: nthRoot(x^4, 2) = x^2
+function nthRootExponent(node) {
+  let newNode = clone(node, false);
+
+  const radicandNode = node.args[0];
+  const rootNode = getRootNode(node);
+  const baseNode = radicandNode.args[0];
+  const exponentNode = NodeType.isParenthesis(radicandNode.args[1]) ?
+    radicandNode.args[1].content : radicandNode.args[1];
+  if (rootNode.equals(exponentNode)) {
+    newNode = baseNode;
+    return NodeStatus.nodeChanged(
+      MathChangeTypes.CANCEL_EXPONENT_AND_ROOT, node, newNode);
+  }
+  else if (NodeType.isConstant(rootNode) && NodeType.isConstant(exponentNode)) {
+    const rootValue = parseFloat(rootNode.value);
+    const exponentValue = parseFloat(exponentNode.value);
+    if (rootValue % exponentValue === 0) {
+      const newRootValue = rootValue/exponentValue;
+      const newRootNode = NodeCreator.constant(newRootValue);
+
+      newNode = NodeCreator.nthRoot(baseNode, newRootNode);
+      return NodeStatus.nodeChanged(
+        MathChangeTypes.CANCEL_EXPONENT, node, newNode);
+    }
+    else if (exponentValue % rootValue === 0) {
+      const newExponentValue = exponentValue/rootValue;
+      const newExponentNode = NodeCreator.constant(newExponentValue);
+
+      newNode = NodeCreator.operator('^', [baseNode, newExponentNode]);
+      return NodeStatus.nodeChanged(
+        MathChangeTypes.CANCEL_ROOT, node, newNode);
+    }
+  }
+
+  return NodeStatus.noChange(node);
+}
+
+// Returns the nthRoot evaluated for a multiplication node.
+// Expects a multiplication node uder the radicand.
+// If the root is a positive constant, it factors the multiplicands, groups them
+// and converts the multiplications into exponents.
+// Distributes the nthRoot into the children nodes, evaluates those nthRoots
```

Comment:
I think this could be easier to follow if the steps were numbered - find it a bit hard to follow still